src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -490,6 +492,24 @@ private PoolEntry createPoolEntry()
                lifetime, MILLISECONDS));
          }
 
+         if (keepaliveTime > 0) {
+            // variance up to 10% of the heartbeat time
+            final long variance = ThreadLocalRandom.current().nextLong(keepaliveTime / 10);
+            final long heartbeatTime = keepaliveTime - variance;
+            poolEntry.setKeepalive(houseKeepingExecutorService.scheduleWithFixedDelay(
+               () -> {
+                  if (connectionBag.reserve(poolEntry)) {
+                     if (!isConnectionAlive(poolEntry.connection)) {
+                        softEvictConnection(poolEntry, DEAD_CONNECTION_MESSAGE, true);
+                        addBagItem(connectionBag.getWaitingThreadCount());
+                     }
+                     else {
```

Comment:
I forgot there should be a call to ``connectionBag.unreserve(poolEntry)`` here, in the ``else`` case.